### PR TITLE
DND-706: Update fork to the latest version and make build automated

### PR DESCRIPTION
**Pull request by spotoczny**
This PR updates the fork to the latest version and applies SmartRecruiters customization
Adds GitHubFLow to automate plugin build process

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -65,12 +65,25 @@
...
+            <artifactId>credentials</artifactId>
+            <version>1214.v1de940103927</version>
+        </dependency>
     </dependencies>
 
     <properties>
-        <revision>1.10</revision>
-        <changelist>-SNAPSHOT</changelist>
-        <jenkins.version>2.361.4</jenkins.version>
+        <jenkins.version>2.375</jenkins.version>
```
Comment (kasinskim) [inline pom.xml:86]: That's the version on `green` will this be compatible?
```
 2.387.1
```

File: src/test/java/org/jenkinsci/plugins/builduser/BuildUserTest.java [hunk 2]
```diff
@@ -87,16 +89,16 @@ public void testMakeUserBuildVariablesWithoutUpstream() throws Exception {
             }
         }
         Assert.assertNotNull("Cannot extract the UpstreamCause", upstreamCause);
-        buildUser.makeBuildVariables(downstreamBuild, outputVars); // Just a smoke check
-        
+        //        buildUser.makeBuildVariables(downstreamBuild, outputVars); // Just a smoke check
```
Comment (kasinskim) [inline src/test/java/org/jenkinsci/plugins/builduser/BuildUserTest.java:92]: ?
```
        //        buildUser.makeBuildVariables(downstreamBuild, outputVars); // Just a smoke check
```

File: src/test/java/org/jenkinsci/plugins/builduser/BuildUserTest.java [hunk 3]
```diff
@@ -87,16 +89,16 @@ public void testMakeUserBuildVariablesWithoutUpstream() throws Exception {
...
+
         // Delete master build and check the correctness
         upstreamBuild.delete();
         try {
-            buildUser.makeBuildVariables(downstreamBuild, outputVars);
+            //            buildUser.makeBuildVariables(downstreamBuild, outputVars);
```
Comment (kasinskim) [inline src/test/java/org/jenkinsci/plugins/builduser/BuildUserTest.java:97]: ?
```
            //            buildUser.makeBuildVariables(downstreamBuild, outputVars);
```

File: pom.xml (hunk 1 above)
Comment (spotoczny) [reply pom.xml:86]: Jest it is.

File: src/test/java/org/jenkinsci/plugins/builduser/BuildUserTest.java (hunk 2 above)
Comment (spotoczny) [reply src/test/java/org/jenkinsci/plugins/builduser/BuildUserTest.java:92]: It was cherry-picked.

File: pom.xml (hunk 1 above)
Comment (spotoczny) [reply pom.xml:86]: Yes, it is. I have upgraded it a little so it conforms with a gh pr label plugin for which the latest dependencies forced to update the jenkins lib to at least version 2.375

**Review by vanta — APPROVED** [review]

**Pull request merged**